panel/models/json.ts

```diff
@@ -0,0 +1,59 @@
+import * as p from "@bokehjs/core/properties"
+import {Markup} from "@bokehjs/models/widgets/markup"
+import JSONFormatter from "json-formatter-js"
+import {PanelMarkupView} from "./markup"
+
+export class JSONView extends PanelMarkupView {
+  model: JSON
+
+  render(): void {
+    super.render();
+    const text = this.model.text.replace(/(\r\n|\n|\r)/gm, "").replace("'", '"')
+    let json;
+    try {
+      json = window.JSON.parse(text)
+    } catch(err) {
+      this.markup_el.innerHTML = "<b>Invalid JSON:</b> " + err.toString()
+      return
+    }
+    const config = {hoverPreviewEnabled: this.model.hover_preview, theme: this.model.theme}
+    const formatter = new JSONFormatter(json, this.model.depth, config)
+    const rendered = formatter.render()
+    let style = "border-radius: 5px; padding: 10px;";
+    if (this.model.theme == "dark")
+      rendered.style.cssText = "background-color: rgb(30, 30, 30);" + style;
+    else
+      rendered.style.cssText = style;
+	this.markup_el.append(rendered)
```

Comment:
Bad indentation or missing `{}`?

Reply:
Bad indentation, thanks!

Reply:
I need this finish linting support in the extensions' compiler, then this kind of error can be more easily detected.

Reply:
I also really need to set up emacs in a way that it doesn't use tabs for ts files...should have done that a long time ago.